spiffworkflow/variables.tf

```
@@ -0,0 +1,126 @@
+variable "cf_org_name" {
+  type        = string
+  description = "cloud.gov organization name"
+}
+
+variable "cf_space_name" {
+  type        = string
+  description = "cloud.gov space in which to deploy the apps"
+}
+
+variable "app_prefix" {
+  type        = string
+  description = "prefix to use for the three application names (<app_prefix>-[connector|backend|frontend])"
+  default     = "spiffworkflow"
+}
+
+variable "route_prefix" {
+  type        = string
+  description = "prefix to use for the application routes (<route_prefix>-connector.app.internal, <route_prefix>.apps.fr.cloud.gov[/api]); leave empty to generate"
+  default     = ""
+}
+
+variable "rds_plan_name" {
+  type        = string
+  description = "PSQL database service plan to use"
+  # See options at https://cloud.gov/docs/services/relational-database/#plans
+  default = "small-psql"
+}
+
+variable "rds_json_params" {
+  description = "A JSON string of arbitrary parameters"
+  type        = string
+  default     = null
+  # See options at https://cloud.gov/docs/services/relational-database/#setting-optional-parameters-1
+}
+
+variable "tags" {
+  description = "A list of tags to add to the module's resource"
+  type        = set(string)
+  default     = []
+}
+
+variable "process_models_repository" {
+  type        = string
+  description = "git repository with process models (for read-write, use SSH-style 'git@github.com:...' and supply your ssh_key)"
+}
+
+variable "process_models_ssh_key" {
+  type        = string
+  description = "private SSH key (only needed for read-write to SSH-style 'git@github.com:...' repositories)"
```

[question] should this have a `default = null` if it's only needed for ssh-style repos? How would you configure a read-only repo?

---

The ssh key is responsible for not only cloning the repo down to load the "example" process models, but also commiting back to it with changes. For example, editing a .bpmn directly on the application, directly commits it back into the target branch, to allow a PR to merge back into main for a permanent update to said models.

Not entirely sure this supports read only, as someone with admin rights on the app can directly edit the bpmn, and will throw an error when saving the model if it cannot commit back to the repo. 

To that point, we can't specify a null value for the var, as a ssh key is a requirement for loading the initial models from https://github.com/gsa-TTS/gsa-process-models via git ssh. It will need to have a value for full functionality.

---

[suggestion] to avoid confusion about whether read-write is optional:

```suggestion
  description = "private SSH key with read/write access to var.process_models_repository repo"
```

---

Done!